### PR TITLE
Add Azure gems to README

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: activestorage/README.md [hunk 1]
```diff
@@ -88,7 +88,8 @@ Variation of image attachment:
 1. Run `rails activestorage:install` to create needed directories, migrations, and configuration.
 2. Optional: Add `gem "aws-sdk", "~> 2"` to your Gemfile if you want to use AWS S3.
 3. Optional: Add `gem "google-cloud-storage", "~> 1.3"` to your Gemfile if you want to use Google Cloud Storage.
-4. Optional: Add `gem "mini_magick"` to your Gemfile if you want to use variants.
+4. Optional: Add `gem 'azure-core', git: "https://github.com/dixpac/azure-ruby-asm-core.git"` and `gem 'azure-storage', require: false` to your Gemfile if you want to use Microsoft Azure.
```
Comment (georgeclaghorn) [inline activestorage/README.md]: Minor nit: use double quotes to match the other optional steps.

Comment (ffmike) [inline activestorage/README.md]: Yeah, good point.

Also at some point presumably we can switch to the release version of the azure-core gem. I'm torn between wanting to note that or not, it's one more thing to keep track of for the future.

Comment (georgeclaghorn) [inline activestorage/README.md]: Looks like @dixpac’s PR (Azure/azure-ruby-asm-core#39) was merged. @dixpac, can we switch back to the released gem now?

File: activestorage/README.md [hunk 2]
```diff
@@ -88,7 +88,8 @@ Variation of image attachment:
 1. Run `rails activestorage:install` to create needed directories, migrations, and configuration.
 2. Optional: Add `gem "aws-sdk", "~> 2"` to your Gemfile if you want to use AWS S3.
 3. Optional: Add `gem "google-cloud-storage", "~> 1.3"` to your Gemfile if you want to use Google Cloud Storage.
-4. Optional: Add `gem "mini_magick"` to your Gemfile if you want to use variants.
+4. Optional: Add `gem "azure-core", git: "https://github.com/dixpac/azure-ruby-asm-core.git"` and `gem "azure-storage", require: false` to your Gemfile if you want to use Microsoft Azure.
```
Comment (schneems) [inline activestorage/README.md]: Why doesn't the released version work here? Any idea when master will be released in a new gem?

Comment (georgeclaghorn) [inline activestorage/README.md]: The released gem didn't support StringIO uploads. It does as of yesterday.

Comment (ffmike) [inline activestorage/README.md]: Just confirmed in my sample project that the release gem is working fine. Updated the PR accordingly.

File: activestorage/README.md [hunk 3]
```diff
@@ -88,7 +88,8 @@ Variation of image attachment:
 1. Run `rails activestorage:install` to create needed directories, migrations, and configuration.
 2. Optional: Add `gem "aws-sdk", "~> 2"` to your Gemfile if you want to use AWS S3.
 3. Optional: Add `gem "google-cloud-storage", "~> 1.3"` to your Gemfile if you want to use Google Cloud Storage.
-4. Optional: Add `gem "mini_magick"` to your Gemfile if you want to use variants.
+4. Optional: Add `gem "azure-core"` and `gem "azure-storage", require: false` to your Gemfile if you want to use Microsoft Azure.
```
Comment (kamipo) [inline activestorage/README.md]: We don't need to add `gem "azure-core"` once #30041 has merged.

File: activestorage/README.md [hunk 4]
```diff
@@ -88,7 +88,7 @@ Variation of image attachment:
 1. Run `rails activestorage:install` to create needed directories, migrations, and configuration.
 2. Optional: Add `gem "aws-sdk", "~> 2"` to your Gemfile if you want to use AWS S3.
 3. Optional: Add `gem "google-cloud-storage", "~> 1.3"` to your Gemfile if you want to use Google Cloud Storage.
-4. Optional: Add `gem "azure-core", git: "https://github.com/dixpac/azure-ruby-asm-core.git"` and `gem "azure-storage", require: false` to your Gemfile if you want to use Microsoft Azure.
+4. Optional: Add `gem "azure-core"` and `gem "azure-storage", require: false` to your Gemfile if you want to use Microsoft Azure.
```
Comment (georgeclaghorn) [inline activestorage/README.md]: The  `azure-storage` gem depends on `azure-core`, so we need not manually bundle the latter.

Comment (georgeclaghorn) [inline activestorage/README.md]: Let's omit `require: false` to match the other optional steps. It's not strictly necessary.

**Comment by dixpac** [comment]
We can switch to the official release as of yesterday. I've submitted the PR yesterday #30041

**Comment by schneems** [comment]
Great, thanks!

**Pull request merged**